### PR TITLE
Fix the synId overflow test in kernel

### PR DESCRIPTION
**Pull request by tingshuc**
The maximum number of synapses per neuron supported by current kernel is 2^16, which is 65536. This fix changes synId (synapse id) overflow test in the kernel. Also, this fix added a test case to test synId overflow.

### CODE REVIEW

File: carlsim/test/core.cpp [hunk 1]
```diff
@@ -636,3 +636,18 @@ TEST(Core, saveLoadSimulation) {
 		}
 	}
 }
+
+TEST(Core, synapseIdOverflow) {
+	::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+	CARLsim sim("Core.synapseIdOverflow", HYBRID_MODE, SILENT, 0, 42);
+
```
Comment (mbeyeler) [inline carlsim/test/core.cpp]: Is it that you have 16bits to address synapses, so having 2^16-1 synapses is fine, but starting at 2^16 it should break? If so, I would suggest creating 65,536 neurons in `gExc` instead of 80,000 to test the corner case.

Comment (tingshuc) [inline carlsim/test/core.cpp]: Thanks, Mike, it is done
